roles/setup-azure-security/tasks/main.yml

```diff
@@ -1,122 +1,113 @@
-- name: Enable Microsoft Defender
-  azure_rm_deployment:
-    resource_group: "{{ resource_group_name }}"
-    location: "{{ azure_region }}"
-    name: Defender
-    template:
-      $schema: "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#"
-      contentVersion: "1.0.0.0"
-      resources:
-        - type: "Microsoft.Security/defenderForStorageSettings"
-          apiVersion: "2022-12-01-preview"
-          name: "current"
-          scope: "Microsoft.Storage/storageAccounts/{{ storage_account_name }}"
-          properties:
-            isEnabled: true
-            malwareScanning:
-              onUpload:
-                isEnabled: true
-                capGBPerMonth: 5000
-            sensitiveDataDiscovery:
-              isEnabled: true
-            overrideSubscriptionLevelSettings: true
-  when: false
+## Defender will typically be enabled at the subscription level. To handle Malware Scanning, we create a basic Logic App here.
+# This will delete any files that are uploaded to the storage account that are detected as malware, however, it will not provide notification of the event. This can be handled within Defender if we have permissions, or by creating a more complex Logic App that logs to Slack.
+# Once created, the logic app should be manually linked to Malware events in the Defender portal. See documentation for more information.
+- set_fact:
+    malware_logic_app_name: "{{ alphanumeric_resource_name_prefix }}-LOGICAPP-Remove-MalwareBlob"
 
-# Firewall
-- name: Create a subnet for Firewall
-  azure_rm_subnet:
-    resource_group: "{{ resource_group_name }}"
-    virtual_network_name: "{{resource_prefix}}-v-net"
-    name: AzureFirewallSubnet
-    private_endpoint_network_policies: "Disabled" # https://learn.microsoft.com/en-us/azure/private-link/disable-private-link-service-network-policy?tabs=private-link-network-policy-cli
-    private_link_service_network_policies: "Disabled"
-  register: subnet_output
+- name: Create Logic App for Malware Scanning
+  command: >
+    az deployment group create \
+      --name MalwareRemovalDeployment \
+      --resource-group {{ resource_group_name }} \
+      --template-file {{ role_path }}/remove-malware-template.bicep \
+      --parameters defender_logic_app_malware_removal_name={{ malware_logic_app_name }} \
+      --parameters storage_account_name={{ storage_account_name }} \
+      --parameters location={{ azure_region }}
 
-- name: Create a subnet for Firewall Management
-  azure_rm_subnet:
-    resource_group: "{{ resource_group_name }}"
-    virtual_network_name: "{{resource_prefix}}-v-net"
-    name: AzureFirewallManagementSubnet
-    private_endpoint_network_policies: "Disabled" # https://learn.microsoft.com/en-us/azure/private-link/disable-private-link-service-network-policy?tabs=private-link-network-policy-cli
-    private_link_service_network_policies: "Disabled"
-  register: management_subnet_output
+# # Firewall
```

Comment:
Everything below this line is commented out?

Reply:
>(Also comments out the outgoing firewall stuff as that's not used anywhere yet and deploying it would cause chaos)